packages/mongodb-test-server/bin/testserver.js

```diff
@@ -0,0 +1,286 @@
+  }
+
+  console.info(`\n${url} available.`);
+}
+
+function assertPortOpen(port) {
+  return new Promise((resolve, reject) => {
+    const client = new net.Socket();
+    client.setTimeout(100);
+    client.on('connect', function () {
+      console.debug('connect', port, 'succeeded');
+      client.end();
+      resolve();
+    });
+
+    client.on('timeout', function () {
+      if (isDebug) {
+        console.debug('Timeout, port should be closed.');
+      }
+
+      client.destroy();
+      reject(new Error(`port ${port} is closed`));
+    });
+
+    client.on('error', function (err) {
+      if (isDebug) {
+        console.debug('port is closed.', err);
+      }
+
+      try {
+        client.destroy();
+      } catch (e) {
+        //
+      }
+
+      reject(err);
+    });
+
+    client.connect(port, 'localhost');
+  });
+}
+
+async function assertPortClosed(port) {
+  const isOpen = await assertPortOpen(port)
+    .then(() => true)
+    .catch(() => false);
+
+  if (isOpen) {
+    throw new Error(`port ${port} is open`);
+  }
+}
+
+async function start(instanceName, options) {
+  const yargsOptions = yargsParser(options);
+  const port = +(yargsOptions.port ?? (await getPort()));
+
+  // make sure other instances are not running on the same port
+  // before doing anything
+  await waitForStopped(port);
+  initializeLogDir();
+  initializePidsDir(instanceName, port);
+
+  const instanceDataPath = getInstanceDataPath(instanceName, port);
+
+  // nuke data from previous runs:
+  await rimraf(instanceDataPath);
+  writePort(instanceName, port);
+
+  const mongodbRunnerArgs = [
+    'start',
+    `--dbpath=${getInstanceDataPath(instanceName, port)}`,
+    `--logpath=${getInstanceLogPath(instanceName, port)}`,
+    `--port=${port}`,
+    `--pidpath=${getInstancePidsPath(instanceName, port)}`,
+    ...options.filter(
+      (opt, i) => !(opt.startsWith('--port') || options[i - 1] === '--port')
+    ),
+  ];
+
+  const mongodbRunnerPath = getMongodbRunnerPath();
+  console.info('Running', mongodbRunnerPath, mongodbRunnerArgs);
+  crossSpawn.sync(mongodbRunnerPath, mongodbRunnerArgs, { stdio: 'inherit' });
+  await waitForRunning(port);
+}
+
+async function stop(instanceName) {
+  const port = readPort(instanceName);
+  const mongodbRunnerWorkerPids = readWorkerPids(instanceName, port);
+  const serverPids = readServerPids(instanceName, port);
+
+  try {
+    await assertPortOpen(port);
+  } catch (err) {
+    console.info(
+      'Server port is already closed, the server has been killed or crashed',
+      err
+    );
+  }
+
+  const processes = [...mongodbRunnerWorkerPids, ...serverPids];
+  await sendSignalToAll(processes, 'SIGKILL');
+  await waitForStopped(port);
+}
+
+async function sendSignalToAll(pids, sig) {
+  const runningProcesses = pids.filter(isRunning);
+
+  if (runningProcesses.length) {
+    for (const pid of runningProcesses) {
+      try {
+        console.log(`sending ${sig} to`, pid);
+        process.kill(pid, sig);
```

Comment:
I'm curious: killing processes by using process.kill wasn't working on windows, this is why we have this code in killall-mongo.js script

https://github.com/mongodb-js/compass/blob/3da3fa3dabcdebc3b07dc8ca88b11cd8fa42d116/scripts/killall-mongo.js#LL25C23-L25C31

Is it not the issue anymore?